include/dlaf/matrix/layout_info.h

```diff
@@ -17,29 +17,39 @@ namespace dlaf {
 namespace matrix {
 class LayoutInfo {
 public:
-  LayoutInfo(GlobalElementSize size, TileElementSize block_size, SizeType tile_ld,
+  LayoutInfo(const GlobalElementSize& size, const TileElementSize& block_size, SizeType tile_ld,
              std::size_t tile_offset_row, std::size_t tile_offset_col);
```

Comment:
Documentation will need to be rewritten when we distribute the matrix.
Therefore I didn't add the documentation to avoid duplicate work.

I may add a placeholder for that if needed.